flopy/modflow/mfflwob.py

```diff
@@ -314,3 +341,183 @@ def write_file(self):
         # swm: END hack for writing standard file
 
         return
+
+    @staticmethod
+    def load(f, model, flowtype=None, ext_unit_dict=None, check=True):
+        """
+        Load an existing package.
+
+        Parameters
+        ----------
+        f : filename or file handle
+            File to load.
+        model : model object
+            The model object (of type :class:`flopy.modflow.mf.Modflow`) to
+            which this package will be added.
+        ext_unit_dict : dictionary, optional
+            If the arrays in the file are specified using EXTERNAL,
+            or older style array control records, then `f` should be a file
+            handle.  In this case ext_unit_dict is required, which can be
+            constructed using the function
+            :class:`flopy.utils.mfreadnam.parsenamefile`.
+        check : boolean
+            Check package data for common errors. (default True)
+
+        Returns
+        -------
+        flwob : ModflowFlwob package object
+            ModflowFlwob package object.
+
+        Examples
+        --------
+
+        >>> import flopy
+        >>> m = flopy.modflow.Modflow()
+        >>> hobs = flopy.modflow.ModflowFlwob.load('test.drob', m)
+
+        """
+
+        if model.verbose:
+            sys.stdout.write('loading flwob package file...\n')
+
+        if not hasattr(f, 'read'):
+            filename = f
+            f = open(filename, 'r')
+
+        ext = None
+        if flowtype is None:
+            # attempt to infer flowtype
+            ext = f.name.split('.')[-1].lower()
+            if 'ch' in ext.lower():

```

Comment:
I am not sure we can rely on the file extension to determine the flow type. I think `ext_unit_dict` has the `fname`'s in them maybe the extension can be a backup if `ext_unit_dict` is not passed in.

Reply:
The PR looks good. Could you also add an existing ghb observation file to the example in `examples/data/mf2005_obs` so that we can have a test that 1) copies the existing files to a new work directory and 2) runs the existing files then 3) loads the existing model, 4) writes the files to another directory, and then 5) runs the new model to make sure MODFLOW will run with a flwobs file created by flopy. That way we know if we ever break the functionality. Maybe this test is there but I missed it (I didn't see any flwobs in the `mf2005_test` directory.

Reply:
There is a river observation file, `tc1-true.orv`, already in `examples\data\mf2005_obs` that gets copied over for test 041. It gets copied to `\t041\tc1-true\orig`, loaded, run, copied to `\t041\tc1-true\new`, and then run again for `t041_test.py\test_obs_load_and_write()`. I can see that it's being copied and run from the `new` directory since the package is included in the namefile for the new copy of the model and there is output in `tc1-true.orv.out`. Do you want an additional test specifically for ghb or is this sufficient?

Reply:
No the river observation is good - I missed it in the name file. What do you think about the issue using the extension to define the `flwobs` type?

Reply:
That sounds fine to me...I'm looking into it now.

Reply:
I added a new global method in `mfflwob.py` to handle the ftype inference, see what you think. I've never passed the `ext_unit_dict` directly into anything so my usage here may not be what you were expecting. Looks like the PY27 test is acting up again.

Reply:
This seems good. I presume you ran the t041 through the debugger to make sure it was doing what is intended (extracting `fname` and is preserving the original input and output unit numbers for the river observations).

I restarted py27 so should pass.

I think we are nearly there. Thanks for hanging in there.

Reply:
Yes, the loader seems to preserve unit numbers when loading.

Reply:
Actually, let me qualify that a bit—it looks like the unit number of the obs package file is preserved, but the output data file unit number is over-ridden during the creation of the new `ModflowFlwob` object according to the `outunits` dictionary that assigns the unit for output files based on the flowtype.
https://github.com/jbellino-usgs/flopy/blob/df3f5e276fe1c0dcbcff1b41df3dcc660842b791/flopy/modflow/mfflwob.py#L145-L148